docs/DesignDocs/AttestationQPLEnforcementSupport.md

```diff
@@ -0,0 +1,86 @@
+Attestation QPL Enforcement Support
+====
+
+This proposal is to make a change to `oe_verify_evidence`, so that it can support policy based QPL (Quote Provider Library) enhancement, such as allowing the API user to pass data to QPL for more advanced quote verification.
+
+Motivation
+----
+
+Intel SGX rolls out PCK certificate and collateral information about every six month. There is a scenario that a quote is signed by the last version of PCK certificate cannot be verified with the latest version of collateral information. This causes unnecessary attestation failures when new versions of PCK certificate and collateral information are rolled out. To deal with this issue, the QPL need to be able to return any version collateral information based on additional query parameters (referred as baselines). With that, a new API called `sgx_ql_get_quote_verification_collateral_with_params` is introduced in QPL. Changes are also needed at OE SDK to call this new API.
+
+Goals:
+ - Verifier can call `oe_verify_evidence` and pass QPL specific parameters to do more advanced quote verification.
+
+User Experience
+----
+
+A new policy type `OE_POLICY_SGX_COLLATERAL_BASELINE` is added to current policy types. Quote verifier can pass multiple policies with this type, while OE SDK would pass all the policies down to the QPL API without interpreting the data content.
+
+Specification
+----
+
+### New policy type & its data structure
+
+- `OE_POLICY_SGX_COLLATERAL_BASELINE` in `oe_policy_type_t`
+- `oe_sgx_collateral_query_param`, which is defined as below,
+```C
+// defined by QPL
+#define MAX_PARAM_STRING_SIZE (255)
+typedef struct _oe_sgx_collateral_query_param
+{
+uint8_t key[MAX_PARAM_STRING_SIZE+1];
+uint8_t value[MAX_PARAM_STRING_SIZE+1];
+} oe_sgx_collateral_query_param_t;
+```
+
+### New APIs / OCalls
+
+Though there is no API signature changes for `oe_verify_evidence`, some internal APIs that potentially get used by tools need to be changed. For backward compatibility, the following new APIs will be added to ensure all existing code either build from OE SDK headers and libraries or source directly can still work,
+- `oe_get_sgx_endorsements_with_policies`, after change, this API will be called by existing `oe_get_sgx_endorsements` without any policy. This new API will not be exposed to OE SDK users.
+- `oe_get_quote_verification_collateral_with_params_ocall`, the current API is `oe_get_quote_verification_collateral_ocall` is an API defined in sgx/attestation.edl, which could have been used by user enclaves, so need to bring this new API to work with existing API.
+
+### Code example
+
+With the new policy, the OE SDK client will be able to pass in multiple collateral baseline parameters as below,
+```C
+oe_sgx_collateral_query_param_t* baselines = (oe_sgx_collateral_query_param_t*)malloc(sizeof(oe_sgx_collateral_query_param_t) * 2);
+memcpy_s(baselines[0].key, MAX_PARAM_STRING_SIZE + 1, "key1", strlen("key1") + 1);
```

Comment:
What are the legal keys?   I'm guessing "key1" is not a real key name.

Reply:
I have updated the code to two sample keys from MAA